src/extractor/edge_based_graph_factory.cpp

```diff
@@ -638,6 +599,398 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
+
+        if (turn_is_valid && (only_restriction_to_node != SPECIAL_NODEID) &&
+            (to_node != only_restriction_to_node))
+        {
+            // We are at an only_-restriction but not at the right turn.
+            ++restricted_turns_counter;
+            turn_is_valid = false;
+        }
+
+        if (turn_is_valid)
+        {
+            if (is_barrier_node)
+            {
+                if (from_node != to_node)
+                {
+                    ++skipped_barrier_turns_counter;
+                    turn_is_valid = false;
+                }
+            }
+            else
+            {
+                if (from_node == to_node && m_node_based_graph->GetOutDegree(turn_node) > 1)
+                {
+                    auto number_of_emmiting_bidirectional_edges = 0;
+                    for (auto edge : m_node_based_graph->GetAdjacentEdgeRange(turn_node))
+                    {
+                        auto target = m_node_based_graph->GetTarget(edge);
+                        auto reverse_edge = m_node_based_graph->FindEdge(target, turn_node);
+                        if (!m_node_based_graph->GetEdgeData(reverse_edge).reversed)
+                        {
+                            ++number_of_emmiting_bidirectional_edges;
+                        }
+                    }
+                    if (number_of_emmiting_bidirectional_edges > 1)
+                    {
+                        ++skipped_uturns_counter;
+                        turn_is_valid = false;
+                    }
+                }
+            }
+        }
+
+        // only add an edge if turn is not a U-turn except when it is
+        // at the end of a dead-end street
+        if (m_restriction_map->CheckIfTurnIsRestricted(from_node, turn_node, to_node) &&
+            (only_restriction_to_node == SPECIAL_NODEID) && (to_node != only_restriction_to_node))
+        {
+            // We are at an only_-restriction but not at the right turn.
+            ++restricted_turns_counter;
+            turn_is_valid = false;
+        }
+
+        // unpack first node of second segment if packed
+
+        const auto first_coordinate =
+            getRepresentativeCoordinate(from_node, turn_node, via_edge, INVERT);
+        const auto third_coordinate =
+            getRepresentativeCoordinate(turn_node, to_node, onto_edge, !INVERT);
+
+        const auto angle = util::coordinate_calculation::computeAngle(
+            first_coordinate, m_node_info_list[turn_node], third_coordinate);
+
+        const auto turn = AnalyzeTurn(from_node, via_edge, turn_node, onto_edge, to_node, angle);
+
+        auto confidence = getTurnConfidence(angle, turn);
+        if (!turn_is_valid)
+            confidence *= 0.8; // makes invalid turns more likely to be resolved in conflicts
+
+        turn_candidates.push_back({onto_edge, turn_is_valid, angle, turn, confidence});
+    }
+    using std::sort;
+    using std::begin;
+    using std::end;
```

Comment:
I don't think you need any of those, the type you're sorting is a `std::vector`, so `begin/end(vec)` finds `std::begin/end` due to argument dependent lookup (ADL), then you can call `std::sort` on those iterators.